sched/jobDefSerializer_test.go

```diff
@@ -0,0 +1,249 @@
+package sched
+
+import (
+	"testing"
+	//"github.com/leanovate/gopter"
+	//"github.com/leanovate/gopter/prop"
+	"fmt"
+	"time"
+	"github.com/leanovate/gopter"
+	"github.com/leanovate/gopter/prop"
+)
+
+const serializeType = BinarySerialize
+
+
+func Test_JobDefCompare(t *testing.T) {
+	origJobDef := makeSampleJobDefintion()
+
+	// validate comparison logic
+	if ok, msg := origJobDef.Equal(origJobDef); !ok {
+		panic(fmt.Sprintf("error: comparison logic doesn't work comparing an object against itself %s\n", msg))
+	}
+	var actualJobDef JobDefinition
+	if ok, msg := origJobDef.Equal(actualJobDef); ok {
+		panic(fmt.Sprintf("error: comparison logic doesn't work comparing an object against nil %s\n", msg))
+	}
+
+}
+
+func Test_NilJobDefinition(t *testing.T) {
+	if ok, err := Serialize(nil, serializeType); err == nil {
+		panic("error: should have gotten an error from serializing a nil object\n")
+		fmt.Printf(string(ok))
```

Comment:
nit: would remove the print statement, typically if tests pass there isn't output